### PR TITLE
:astronaut: Physics engine utils: movement methods

### PR DESCRIPTION
**Pull request by landryarki**
add a Vector3 type in Physics class

add multiple ApplyMovement method to handle: position, velocity, acceleration and terminal velocitiy, all over time

add other method to apply specific type of force to an object.

### CODE REVIEW

**Comment by landryarki** [comment]
Should i leave the declaration of Vector3 here ? or should i create a Math class with a Vector Class instead of a struct ? or use the sfml Vector3 ?

**Comment by landryarki** [comment]
also the method with a terminal velocity ideally shouldn't exist and the terminal velocity would naturally come of as we apply all resistance forces to the object. But i added it for the prototype

**Comment by landryarki** [comment]
> > also the method with a terminal velocity ideally shouldn't exist and the terminal velocity would naturally come of as we apply all resistance forces to the object. But i added it for the prototype
> 
> I'm sorry, I don't understand what part of the code you're talking about. Can you please indicate the lines concerned?

Physics.cpp l.30 to 46 the method brut force a terminal velocity but it really shouldn't exist in a more advanced engine where friction and resistance are emulated. but for now it is here

File: src/physics/Math.hpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
+/*
+** EPITECH PROJECT, 2024
+** Project
+** File description:
+** No file there , just an epitech header example .
```
Comment (AubaneNourry) [inline src/physics/Math.hpp:5]: can you update your header?

**Review by AubaneNourry — APPROVED** [review]
:+1: , just a little forgotten header

**Review by 6im0n — APPROVED** [review]
Good for me ! :ok:

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by RenardFute — APPROVED** [review]

**Pull request merged**